### PR TITLE
Added frontend tests

### PR DESCRIPTION
**Pull request by ArneTR**
A long time overdue, but now finally ready: The frontend tests!

- Loading demo data
- Accessing frontend through Firefox Browser
- Checking functionality and some sample values

### CODE REVIEW

**Comment by ArneTR** [comment]
@ribalba Any wishes for more frontend tests?

File: .github/actions/gmt-pytest/action.yml [hunk 1]
```diff
@@ -44,6 +44,7 @@ runs:
       run: |
         source venv/bin/activate
         python3 -m pip install -r requirements-dev.txt
+        playwright install --with-deps firefox
```
Comment (ribalba) [inline .github/actions/gmt-pytest/action.yml:47]: In the description you write with Chrome and here you also add Firefox

File: tests/frontend/test_frontend.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+#pylint: disable=unused-argument
+@pytest.fixture(autouse=True, scope='module')
+def setup_and_cleanup_module():
+    # before
+
+    global page #pylint: disable=global-statement
+    global context #pylint: disable=global-statement
+    # start only one browser for whole file
+    playwright = sync_playwright().start()
+    browser = playwright.firefox.launch()
```
Comment (ribalba) [inline tests/frontend/test_frontend.py:25]: firefox again

**Review by ribalba — APPROVED** [review]
There can always be more tests but this looks quite neat for a first version

File: tests/frontend/test_frontend.py (hunk 2 above)
Comment (ArneTR) [reply tests/frontend/test_frontend.py:25]: thanks. description is wrong. Firefox is what I want

File: .github/actions/gmt-pytest/action.yml (hunk 1 above)
Comment (ArneTR) [reply .github/actions/gmt-pytest/action.yml:47]: thanks. description is wrong. Firefox is what I want

**Comment by ArneTR** [comment]
@ribalba Please merge in when review done

**Comment by ribalba** [comment]
@ArneTR the tests are failing because of a wrong value?

**Comment by ArneTR** [comment]
Oh I see, you merged in main ... will check what change that introduced

**Comment by ArneTR** [comment]
There were some time defaults in the tests which would flake when time advances. I now set all time boxes with manually selected fields

**Pull request merged**